bleak/backends/bluezdbus/client.py

```diff
@@ -904,6 +916,29 @@ async def write_gatt_descriptor(
             "Write Descriptor %s | %s: %s", descriptor.handle, descriptor.obj[0], data
         )
 
+    def _read_notify_fd(self, fd, callback):
+        loop = asyncio.get_running_loop()
+
+        def on_data():
+            try:
+                data = os.read(fd, 1024)
+                if not data:
+                    raise RuntimeError("Unexpected EOF on notification file handle")
+                callback(bytes(data))
+            except Exception as e:
+                logger.error(
+                    "AcquireNotify: Read error on fd %d: %s. Notifications have been stopped.",
+                    fd,
+                    e,
+                )
+                try:
+                    loop.remove_reader(fd)
+                    os.close(fd)
+                except OSError:
+                    pass
```

Comment:
```suggestion
                try:
                    loop.remove_reader(fd)
                except RuntimeError:
                    # run loop is closed
                    pass
                try:
                    os.close(fd)
                except OSError:
                    # bad file descriptor
                    pass
```

Reply:
Although doing this here without removing `fd` from `self._notification_fds` is problematic when it comes to later calls.